test/e2e/service.go

```diff
@@ -275,9 +275,6 @@ var _ = Describe("Services", func() {
 			},
 		}
 
-		By("cleaning up previous service " + serviceName + " from namespace " + ns)
```

Comment:
See lines 284-288 - we defer deletion of the service as soon as we see that creation succeeded.

It is still possible that we could leak resources if the cluster is deleted before the service controller has a chance to delete them, but that can't really be avoided without reusing the names.